### ISSUE TITLE
Please remove dependencies on **rgdal**, **rgeos**, and/or **maptools**

### ISSUE THREAD

**Issue by rsbivand** [mablab/rpostgis#27]
This package depends on (depends, imports or suggests) **raster** and one or more of the retiring packages **rgdal**, **rgeos** or **maptools** (https://r-spatial.org/r/2022/04/12/evolution.html). Since **raster** `3.6.3`, all use of external FOSS library functionality has been transferred to **terra**, making the retiring packages very likely redundant. It would help greatly if you could remove dependencies on the retiring packages as soon as possible.

**Comment by basille** [comment]
Thanks @rsbivand for your message. Indeed, `rpostgis` is due a major update to migrate to `sf`/`stars`/`terra`. Unfortunately, I don't have the time or energy to work on this in the foreseeable future. However, given the recent developments of said packages, the question of the actual relevance of `rpostgis` is even open: As far as I know, `sf` can handle the bulk of vector-related functions of the package — it's still not clear if `stars`/`terra` can do the raster counterpart. So is `rpostgis` still useful? It might be time for its retirement altogether.

**Comment by rsbivand** [comment]
The package has a single CRAN dependency **lucas**. While **sf** has a separate DB vector reading architecture, I'm unsure about **terra** and **stars**. You could ask **lucas** whether they need **rpostgis** sufficiently to want to help upgrade. You could also raise an issue on https://github.com/r-spatial/discuss/issues maybe linking to this issue, asking whether anyone could run some comparative checks of vector and raster functionality, so aiming to create a table of alternatives to **rpostgis** workflows.

**Comment by basille** [comment]
Thanks @rsbivand, this is both excellent information and suggestion. It may well have to wait until January so I can free up some time to do it properly though. Depending on the actual need for `rpostgis` in the current R landscape (my evaluation is not that much), it might also be worth asking ROpenSci to get a broader engagement. 

I'll leave this issue open until the actual dependency problem is solved, and I'll open a different one to report updates about the future of `rpostgis`.

**Comment by rsbivand** [comment]
Are we any closer to a resolution? I haven't seen much movement on https://github.com/r-spatial/discuss/issues/58, so maybe your evaluation is realistic. If we are moving to archival, please ask CRAN to archive `rpostgis` before October.

**Comment by basille** [comment]
Unfortunately, nothing really new. Before asking for archival, I'd like to update the package so that a warning is issued on startup. I'll try to do this soon so that the package can be more safely archived in a few months. I'll leave this issue open until then — not hoping for a much different outcome, but simply as a reminder that the problem is still pending.

Thanks @rsbivand for your help and support!

**Comment by rsbivand** [comment]
Maybe move to archiving? I emailed the `lucas` maintainer, perhaps there will be a response?

**Comment by basille** [comment]
Hey @rsbivand, thanks for the update! We've added a startup warning for `rpostgis`, that reads:

```
This is 'rpostgis' version 1.4.4 (2023-05-06).

Due to retirement of the package 'rgeos', 'rpostgis' will retire in September 2023.
  * For vector operations, please check package 'sf', which provides a mechanism to connect to PostGIS databases.
  * For raster operations, no alternative solution is identified yet.
  * For general database operations, use 'RPostgreSQL' directly.

If you are interested in the development and maintenance of 'rpostgis', please check:
https://github.com/mablab/rpostgis/issues/28
```

So we intend to retire the package in a couple of months. Let's see what happens in the meanwhile, but I already contacted `lucas`'s maintainer about 6 months ago, to no avail. (note that we weren't aware that _reading_ rasters from PostGIS works with both `terra` and `stars` so that the warning is partially incorrect)

If that's OK, I'll leave this issue open until `rpostgis` is retired, or until a new maintainer takes it up.

**Comment by rsbivand** [comment]
@basille sounds like a viable plan!

**Comment by rsbivand** [comment]
@basille Less than three weeks remain to fix this.

**Comment by dnbucklin** [comment]
@rsbivand Thanks for the update. We have found a new maintainer (@Cidree), and the current plan is to release an update to address this soon. #28

**Comment by rsbivand** [comment]
@Cidree @dnbucklin @basille R spatial infrastructure packages maptools, rgdal and rgeos will be archived by CRAN on Monday, October 16, 2023. Your package does not pass CMD check when these packages are not available. Expect your package to be archived by CRAN October 17-18 as CRAN checks feed through and your package fails, if not updated by Monday, October 16, 2023. 

No grace period is anticipated, as you have had sufficient time to update your package to remove dependencies on maptools, rgdal and/or rgeos. It remains the case that many packages importing the raster package needlessly depend on retiring packages, as raster stopped using them a year ago.

**Comment by Cidree** [comment]
@rsbivand Thank you for reminding us. The dependencies on `maptools`, `rgdal` and `rgeos` are already removed, and we are fixing the last details. We hope the package will we submitted soon.

**Comment by dnbucklin** [comment]
Closing, as this issue is addressed in `rpostgis` v1.5.

**Comment by rsbivand** [comment]
Thanks for your cooperation!
